quote-book/src/synchronized_quote_book.rs

```diff
@@ -38,6 +38,13 @@ where
         if self.ledger.contains_key_image(&sci.key_image())? {
             return Err(QuoteBookError::QuoteIsStale);
         }
+        let num_blocks = self.ledger.num_blocks()?;
+        if let Some(input_rules) = &sci.tx_in.input_rules {
+            if input_rules.max_tombstone_block != 0 && num_blocks >= input_rules.max_tombstone_block
```

Comment:
I think there is an off by one error here. According to https://github.com/mobilecoinfoundation/mobilecoin/blob/master/transaction/core/src/validation/validate.rs#L450, we treat tombstone block numbers as block indices - The block index at which this transaction is no longer considered valid.
And here, you are comparing number of blocks in ledger to be greater than or equal a block index. I believe that this should compare current_block_index >= tombstone, so you should use `num_blocks - 1` (which could in theory overflow if num blocks is zero, so .checked_sub() can do this safely but realistically I am not sure this is needed here since the server does [ensure the ledger is not empty](https://github.com/mobilecoinofficial/deqs/blob/main/server/src/bin/main.rs#L29) before starting).

Reply:
It is a small optimization but I would re-order the checks to check for tombstone block first since that is cheaper than the key image check, CPU-wise.

Reply:
That's a very good point. Thank you.